app/src/main/java/org/mozilla/rocket/home/HomeViewModel.kt

```diff
@@ -212,28 +212,21 @@ class HomeViewModel(
                 pinTopSiteResult = null
             }
         } else {
-            listOf(SitePage(listOf(Site.EmptyHintSite)))
+            listOf(Site.EmptyHintSite).addDummyTopSites().toSitePages()
         }
     }
 
     private fun List<Site>.toSitePages(): List<SitePage> = chunked(TOP_SITES_PER_PAGE)
             .take(TOP_SITES_MAX_PAGE_SIZE)
             .map { SitePage(it) }
 
-    private fun List<SitePage>.addDummyTopSites(): List<SitePage> {
-        val sitePagesWithDummySite = mutableListOf<SitePage>()
-        this.forEach { sitePage ->
-            if (sitePage.sites.size < TOP_SITES_PER_PAGE) {
-                val sites = sitePage.sites.toMutableList()
-                for (i in 0 until TOP_SITES_PER_PAGE - sitePage.sites.size) {
-                    sites.add(Site.DummySite)
-                }
-                sitePagesWithDummySite.add(SitePage(sites))
-            } else {
-                sitePagesWithDummySite.add(sitePage)
-            }
+    private fun List<Site>.addDummyTopSites(): List<Site> {
+        val siteListWithDummySite = this.toMutableList()
+        val dummySiteSize = TOP_SITES_PER_PAGE * 2 - this.size
```

Comment:
we can replace `2` with `TOP_SITES_MAX_PAGE_SIZE`

Reply:
Sure ! thank you